lib/pure/os.nim

```diff
@@ -250,6 +250,35 @@ proc splitPath*(path: string): tuple[head, tail: string] {.
     result.head = ""
     result.tail = path
 
+proc absolutePrefix*(path: string): string {.since: (1, 1).} =
+  ## returns the root path component for absolute paths, empty for relative paths.
+  ## See also `isAbsolute`.
+  runnableExamples:
+    when defined(posix):
+      doAssert absolutePrefix("//foo") == "//"
+      doAssert absolutePrefix("foo") == ""
+    when defined(windows):
+      doAssert absolutePrefix(r"C:\\\bar") == r"C:\\\"
```

Comment:
But shouldn't this be ``r"C:\"`` instead? There is no need to keep the redundant backslashes.

Reply:
resolved: I'm assuming this comment is resolved because of `Never mind.` in https://github.com/nim-lang/Nim/pull/13265#discussion_r373184796
(i need to keep all DirSeps so that the remainder is a valid relative path and not a wrong absolute path; eg, this is used in `let prefix = path.absolutePrefix` in `nativeToUnixPath`